javalib/src/main/scala/java/util/concurrent/ThreadLocalRandom.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * closely translated from:
+ * https://github.com/scala/scala
+ * scala.concurrent.fokjoin.ThreadLocalRandom
+ */
+
+package java.util.concurrent
+
+import java.util.Random
+
+class ThreadLocalRandom extends Random {
+
+  override def setSeed(seed: Long): Unit =
+    if (initialized) throw new UnsupportedOperationException()
+    else super.setSeed(seed)
+
+
+  def nextInt(least: Int, bound: Int): Int = {
+    if (least >= bound) throw new IllegalArgumentException()
+
+    (nextLong(bound.toLong - least.toLong) + least.toLong).toInt
```

Comment:
Should have a fast path when there is no overflow as per my older comment (which does not show up in the diff anymore).